carbon-p2-plugin/src/main/java/org/wso2/maven/p2/feature/generate/Feature.java

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wso2.maven.p2.feature.generate;
+
+import org.wso2.maven.p2.beans.CarbonArtifact;
+
+/**
+ * Bean class representing a Feature object provided as an input param to FeatureGenMojo
+ */
+public class Feature extends CarbonArtifact {
+    private String id;
+
+    public String getId() {
+        if(id.endsWith(".feature")) {
+            return id.substring(0, id.lastIndexOf(".feature"));
+        }
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public boolean isOptional() {
+        return this.getCompatibility().equals("optional");
```

Comment:
Not necessary for a null check as the method this.getCompatibility() always return a default value. However, this code is also obsolete and no longer exist in the current codebase.